### PR TITLE
filter unit tests

### PR DESCRIPTION
**Pull request by paddykontschak**
## Unit tests

`filter()`, `not()`

## All tests pass

mojave results

### CODE REVIEW

File: tests/cases/dom/traverse/filter.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    }
+);
+
+
+QUnit.test(
+    "filter() on element with specific class",
+    function (assert)
+    {
+        const filter = mojave.dom.traverse.filter;
+        const elements = Array.from(document.querySelectorAll(".test-element-filter"));
```
Comment (apfelbox) [inline tests/cases/dom/traverse/filter.js]: You can use our very own library here.

File: tests/cases/dom/traverse/filter.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+        const result = filter(elements, ".element2");
+
+        assert.equal(result.length, 1, "found 1 element");
+        assert.ok(result[0].classList.contains("element2"), "contains .element2");
+    }
```
Comment (apfelbox) [inline tests/cases/dom/traverse/filter.js]: Could you please add another test, where multiple elements are matched and an additional test where no element is matched, please.

File: tests/cases/dom/traverse/not.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+);
+
+
+QUnit.test(
+    "not() on elements that are children",
+    function (assert)
+    {
+        const not = mojave.dom.traverse.not;
+        const elements = Array.from(document.querySelectorAll(".test-element-not"));
+        const selector = ".test-element-not > :first-child";
```
Comment (apfelbox) [inline tests/cases/dom/traverse/not.js]: That selector doesn't do what you think it does. 😉 
This test seems fragile, we probably should rewrite it using another selector.

File: tests/cases/dom/traverse/not.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+
+QUnit.test(
+    "not() on elements that are children",
+    function (assert)
+    {
+        const find = mojave.dom.traverse.find;
+        const not = mojave.dom.traverse.not;
+        const elements = find(".test-element");
+        const selector = ".test-element-not > :first-child";
```
Comment (apfelbox) [inline tests/cases/dom/traverse/not.js]: The child selector here can be removed

Comment (paddykontschak) [inline tests/cases/dom/traverse/not.js]: Removing the child selector is just resulting in 0 elements found, as you're going through all elements with the class `.test-element` and telling the `not()` function to filter out all elements with the class `.test-element` (`.test-element-not` doesn't exist anymore).

failed

Comment (apfelbox) [inline tests/cases/dom/traverse/not.js]: The test should look something like this:

```js
const elements = find(".test-element");
const filteredElements = not(elements, ".test-element-not");

assert.equal(filteredElements, elements, "no item was filtered as not item was matched for removal");
```

Comment (apfelbox) [inline tests/cases/dom/traverse/not.js]: If that fails, we probably just found a bug in the library.

**Comment by apfelbox** [comment]
Thanks!

**Pull request merged**